src/chem/viirs_aod2ioda.py

```diff
@@ -9,189 +10,236 @@
+                    errs[i] = uncertainty[i]
+
+            #  Write out data
+
+            #  Values
+
+            if first:
+                self.outdata[self.varDict[iodavar]['valKey']] = vals
+                self.outdata[self.varDict[iodavar]['errKey']] = errs
+                self.outdata[self.varDict[iodavar]['qcKey']] = qcall
+            else:
+                self.outdata[self.varDict[iodavar]['valKey']] = np.concatenate(
+                    (self.outdata[self.varDict[iodavar]['valKey']], vals))
+                self.outdata[self.varDict[iodavar]['errKey']] = np.concatenate(
+                    (self.outdata[self.varDict[iodavar]['errKey']], errs))
+                self.outdata[self.varDict[iodavar]['qcKey']] = np.concatenate(
+                    (self.outdata[self.varDict[iodavar]['qcKey']], qcall))
+
+            #  Add Meta data
+
+            if first:
+                self.outdata[('latitude', 'MetaData')] = lats
+                self.outdata[('longitude', 'MetaData')] = lons
+                self.outdata[('datetime', 'MetaData')] = obs_time
+            else:
+                self.outdata[('latitude', 'MetaData')] = np.concatenate((self.outdata[('latitude', 'MetaData')], lats))
+                self.outdata[('longitude', 'MetaData')] = np.concatenate((self.outdata[('longitude', 'MetaData')], lons))
+                self.outdata[('datetime', 'MetaData')] = np.concatenate((self.outdata[('datetime', 'MetaData')], obs_time))
+
+            first = False
+
+        DimDict['nlocs'] = len(self.outdata[('latitude', 'MetaData')])
+        AttrData['nlocs'] = np.int32(DimDict['nlocs'])
+
 
 def main():
 
+    # get command line arguments
+    # Usage: python blah.py -i /path/to/obs/2021060801.nc /path/to/obs/2021060802.nc ... -t Analysis_time /path/to/obs/2021060823.nc
+    # -o /path/to/ioda/20210608.nc
+    # where the input obs could be for any desired interval to concatenated together. Analysis time is generally the midpoint of
+    # analysis window.
     parser = argparse.ArgumentParser(
         description=('Read VIIRS aerosol optical depth file(s) and Converter'
                      ' of native NetCDF format for observations of optical'
-                     ' depth from VIIRS AOD550 to IODA netCDF format.')
+                     ' depth from VIIRS AOD550 to IODA-V2 netCDF format.')
     )
-    parser.add_argument('-i', '--input',
-                        help="name of viirs aod input file(s)",
-                        type=str, required=True)
-    parser.add_argument('-o', '--output',
-                        help="name of ioda output file",
-                        type=str, required=True)
+    parser.add_argument(
+        '-i', '--input',
+        help="path of viirs aod input file(s)",
+        type=str, nargs='+', required=True)
+    parser.add_argument(
+        '-o', '--output',
+        help="name of ioda-v2 output file",
+        type=str, required=True)
+    parser.add_argument(
```

Comment:
Doesn't this come from the file itself? Why does it need specified?

Reply:
The output file is specified by the wrapper code, so it's needed here. There are many input files, and one output file.   The output filename will depend on the analysis time.

Reply:
I'm talking about `-t` as a required argument. It's never actually used as far as I can see...

Reply:
I need to look at this a bit more to see if I can do it without the '-o' argument. It looks like it's being used.  I see it in the other converters (like tropomi), etc.